frontend/src/app/modules/invite-user-modal/principal/principal.component.ts

```diff
@@ -93,13 +147,40 @@ export class PrincipalComponent implements OnInit {
   onSubmit($e:Event) {
     $e.preventDefault();
 
+    if (this.dynamicForm) {
+      this.dynamicForm.validateForm().subscribe(() => {
+        this.onValidatedSubmit();
+      });
+    } else {
+      this.onValidatedSubmit();
+    }
+  }
+
+  onValidatedSubmit() {
     if (this.principalForm.invalid) {
-      this.principalForm.markAsDirty();
       return;
     }
 
+    // The code below transforms the model value as it comes from the dynamic form to the value accepted by the API.
+    // This is not just necessary for submit, but also so that we can reseed the initial values to the payload
+    // when going back to this step after having completed it once.
+    const links = this.customFields!._links || {};
```

Comment:
Wouldn't the dynamic form need this same helper when transforming the model before submission? Can't we reuse that here? It's really specific to have this in this component

Reply:
Yes I agree, and I'm not sure where it would fit best. Perhaps the dynamic form service would be an even better place. To be honest, this feels like something that should just be a pure function and not be tied to any other service or component in particular.